client/src/utils/api-util.js

```diff
@@ -1,107 +1,105 @@
+ * @param url - URL to send the PUT to
+ * @param data - Data to send to the specified URL
+ * @param contentType - The ContentType header, defaults to 'application/json'
+ */
+export async function put(url, data, contentType = 'application/json') {
+    return await sendRequest(url, 'PUT', data, undefined, contentType);
+}
 
-    let res = await fetch(url, {
-        method: 'POST',
-        body: formData
-    })
-        .then(async (res) => {
-            if (!res.ok) {
-                let errorMessage = await res.text();
+/**
+ * Generic function to send an HTTP request to a specified URL
+ * with specified data
+ * @param url - URL to send the request to
+ * @param methodType - String for the type of HTTP request to send
+ *        Defaults to 'GET'
+ * @param data - Data to send to the URL, Defaults to undefined
+ * @param file - File to send to the URL, Defaults to undefined
+ * @param contentType - The ContentType header, defaults to 'application/json'
+ */
+async function sendRequest(url, methodType='GET', data=undefined, file=undefined, contentType = 'application/json') {
+    if (httpMethodTypes.includes(methodType) && url) {
+        let error = null;
+
+        let configuration = {
+            method: methodType
+        };
+
+        if (data) {
+            configuration.body = JSON.stringify(data);
+            configuration.headers = {
+                'Content-Type': contentType
+            };
+        }
+        else if (file) {
+            const formData = new FormData();
+            formData.append('file', file);
 
-                try {
-                    errorMessage = JSON.parse(errorMessage);
-                } catch (error) {}
+            configuration.body = formData;
+        }
 
-                throw Error(errorMessage.message);
-            }
-            return res;
-        })
-        .then(res => res.json())
-        .catch(err => error = err);
-    if (error || res.error) throw(error || res.error);
-    else return res;
+        let res = await fetch(url, configuration)
+            .then(async (res) => {
+                if (!res.ok) {
+                    let errorMessage = await res.text();
+
+                    try {
+                        errorMessage = JSON.parse(errorMessage);
+                    } catch (error) {}
+
+                    throw Error(errorMessage.message);
+                }
+                return res;
+            })
+            .then(res => res.json())
+            .catch(err => error = err);
+        if (error || res.error) throw(error || res.error);
+        else return res;
+    }
 }
 
 /**
  * Uploads the specified images to the monument-images S3 bucket, inside the images/ folder
  * @param images - List of images to upload to S3
  * @returns {Promise<[]>} - Promise that when awaited, returns a List of S3 Object keys for the uploaded images
  */
-export default async function uploadImagesToS3(images) {
+export async function uploadImagesToS3(images) {
```

Comment:
I missed this when it was originally written, but we can't use secret keys on the frontend like this as you could theoretically search the minified production code and find it. We will have to use STS to send temporary credentials to the client from the server: https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/prog-services-sts.html

We can do this in a separate PR. We need to remember to rotate our security keys once it's done as these ones are potentially exposed

Reply:
It might even be better if we just upload the file to our server and let it handle all image uploads/deletes

Reply:
I'm all for us removing all S3 Image uploading from the Front-End. That way it all happens in one place again. We'll have to remember to address that in a PR soon.